src/Microsoft.DotNet.Build.Tasks/Targets/packageresolve.targets

```diff
@@ -85,22 +85,9 @@
 
     <!-- We may have an indirect package reference that we want to replace with a project reference -->
     <ItemGroup>
-      <!-- Convert to filenames so that we can intersect -->
-      <_ProjectReferenceFilenames Include="@(_ResolvedProjectReferencePaths->'%(FileName)%(Extension)')">
-        <OriginalIdentity>%(Identity)</OriginalIdentity>
-      </_ProjectReferenceFilenames>
-
-      <_ReferencesFileNames Include="@(Reference->'%(FileName)%(Extension)')">
-        <OriginalIdentity>%(Identity)</OriginalIdentity>
-      </_ReferencesFileNames>
-
-      <_NoneFileNames Include="@(None->'%(FileName)%(Extension)')">
-        <OriginalIdentity>%(Identity)</OriginalIdentity>
-      </_NoneFileNames>
-      
       <!-- Intersect project-refs with package-refs -->
-      <_ReferenceFileNamesToRemove Include="@(_ReferencesFileNames->'%(OriginalIdentity)')" Condition="'@(_ProjectReferenceFilenames)' == '@(_ReferencesFileNames)' and '%(Identity)' != ''"/>
-      <_NoneFileNamesToRemove Include="@(_NoneFileNames->'%(OriginalIdentity)')" Condition="'@(_ProjectReferenceFilenames)' == '@(_NoneFileNames)' and '%(Identity)' != ''"/>
+      <_ReferenceFileNamesToRemove Include="@(Reference)" Condition="'@(_ResolvedProjectReferencePaths->'%(FileName)%(Extension)')' == '%(FileName)%(Extension)' and '%(Identity)' != ''" />
```

Comment:
@weshaggard Yes, I mentioned in the PR -- xbuild doesn't support metadata references inside the metadata declarations on items, so all those %(OriginalIdentity) references cause NREs in xbuild.